curve25519-dalek/src/edwards.rs

```diff
@@ -567,9 +568,31 @@ impl EdwardsPoint {
         let recip = self.Z.invert();
         let x = &self.X * &recip;
         let y = &self.Y * &recip;
-        let mut s: [u8; 32];
+        Self::compress_affine(x, y)
+    }
 
-        s = y.as_bytes();
+    /// Compress several `EdwardsPoint`s into `CompressedEdwardsY` format, using a batch inversion
+    /// for a significant speedup.
+    #[cfg(feature = "alloc")]
+    pub fn compress_batch(inputs: &[EdwardsPoint]) -> Vec<CompressedEdwardsY> {
+        let mut zs = inputs.iter().map(|input| input.Z).collect::<Vec<_>>();
+        FieldElement::batch_invert(&mut zs);
```

Comment:
Question, more for this entire file: is `Z` ever zero? Why/why not? If it is zero, won't some of these get mapped to `(0, 0)`, which is the 2-torsion point, not the identity?

Reply:
I believe it's not allowed to ever be zero. It's `1` for the additive identity (X:0, Y:1, Z:1, T:0).

https://en.wikipedia.org/wiki/Homogeneous_coordinates suggests `Z` may be `0` for the point at infinity / additive identity, but that isn't the case here.

Reply:
Even if it's identity, `Z` will still be `FieldElement::ONE`.

There's also a test for identity here: https://github.com/dalek-cryptography/curve25519-dalek/pull/759/files#diff-8dd3d09d222437a2e1efde6f6ea33a387ba97798d8dd7db34ee548e33a79587eR2068-R2069